src/components/picker/PickerItem.js

```diff
@@ -34,12 +34,17 @@ const PickerItem = props => {
   const itemValue = !migrate && _.isPlainObject(value) ? value?.value : value;
   const isSelected = isItemSelected(itemValue, context.value);
   const itemLabel = getItemLabel(label, value, props.getItemLabel || context.getItemLabel);
+  const selectedCounter = context.selectionLimit && context.value?.length;
   const accessibilityProps = {
     accessibilityState: isSelected ? {selected: true} : undefined,
     accessibilityHint: 'Double click to select this suggestion',
     ...Modifiers.extractAccessibilityProps(props)
   };
 
+  const isItemDisabled = useMemo(() => {
+    return disabled || (!isSelected && context.selectionLimit && context.selectionLimit === selectedCounter);
+  }, [selectedCounter]);
```

Comment:
Notice you use a simple variable as a dependency in this useMemo. 
Each time the component will render,  `selectedCounter` will be recreated and will invoke the useMemo
Overall it's not that dramatic in this case, but better do it right. 

I'd move `selectedCounter` into the useMemo method and include in the dependencies array values from props/context/state.